packages/canon/src/components/TextField/types.ts

```diff
@@ -54,7 +53,7 @@ export interface TextFieldProps
   /**
    * An icon to render before the input
    */
-  icon?: IconNames;
+  icon?: React.ReactNode;
```

Comment:
Minor, but generally I prefer `import { ReactNode } from 'react'` etc. Especially with the need for the `React` import for JSX gone.